public/js/src/module/connection.js

```diff
@@ -87,7 +146,8 @@ function _uploadBatch( recordBatch ) {
         body: recordBatch.formData
     } )
         .then( response => {
-            const result = {
+            /** @type { UploadBatchResult } */
+            let result = {
```

Comment:
Besides adding types as I go to understand data flow, I’ve tried to also use `let` instead of `const` to identify where a reference is being treated as mutable. There are differing opinions on this, but this has become the most common idiom and can be helpful in understanding how state changes affect data flow.